schlib/rules/S6_2.py

```diff
@@ -134,6 +191,35 @@ def check(self):
             extraFields
             ])
 
+    def check_dcm_file(self):
+        """
+        Proceeds the checking of the rule.
+        The following variables will be accessible after checking:
+            * only_datasheet_missing
+        """
+
+        self.only_datasheet_missing = False
+        invalid_documentation = 0
+
+        # check part itself
+        if self.checkDocumentation(self.component.name, self.component.documentation, False, self.component.isGraphicSymbol() or self.component.isPowerSymbol()):
+            invalid_documentation += 1
+
+        # check all its aliases too
+        if self.component.aliases:
+            invalid = []
+            for alias in self.component.aliases.keys():
+                if self.checkDocumentation(alias, self.component.aliases[alias], True, self.component.isGraphicSymbol() or self.component.isPowerSymbol()):
+                    invalid_documentation += 1
+
+        return invalid_documentation > 0
```

Comment:
No real reason to keep a counter since the result ultimately goes to a Boolean. But I guess in the future it could be handy and it works now too.

Reply:
All other scripts use counters